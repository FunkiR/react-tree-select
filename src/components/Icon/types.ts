import {FontAwesomeIconProps} from '@fortawesome/react-fontawesome';
import {ComponentType} from "react";

export type Props = FontAwesomeIconProps;

export type IconComponent = ComponentType<Props>;
