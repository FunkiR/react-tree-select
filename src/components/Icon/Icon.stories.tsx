import {faCaretDown, faMinusSquare, faPlusSquare,faSearch, faTimesCircle} from "@fortawesome/free-solid-svg-icons";
import {action} from "@storybook/addon-actions";
import {Meta, Story} from '@storybook/react/types-6-0';
import React from 'react';
import {withSelectContext} from "stories/decorators";

import Icon from './Icon';
import type {Props} from './types';

const icons = {faCaretDown, faSearch, faTimesCircle, faMinusSquare, faPlusSquare};

export default {
	title: 'Components/Icon',
	component: Icon,
	decorators: [withSelectContext],
	argTypes: {
		icon: {
			control: {
				type: 'select',
				options: icons
			}
		}
	},
} as Meta;

const Template: Story<Props> = args => <Icon {...args} onClick={action('onClick')} />;

export const Default = Template.bind({});

Default.args = {
	icon: icons.faCaretDown
};

