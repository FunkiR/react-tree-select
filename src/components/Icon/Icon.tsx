import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import cn from 'classnames';
import useSelect from "~/hooks/useSelect";
import React, {FC, memo} from 'react';

import type {Props} from "./types";

export const Icon: FC<Props> = props => {
	const {className, ...iconProps} = props;
	const [theme] = useSelect('theme');

	return <FontAwesomeIcon className={cn(theme.Icon, className)} {...iconProps} />;
};

export default memo(Icon);
