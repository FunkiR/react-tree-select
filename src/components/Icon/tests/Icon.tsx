import {faCaretDown} from "@fortawesome/free-solid-svg-icons";
import Icon from "components/Icon";
import {mount, shallow} from "enzyme";
import React from "react";

describe('Icon component', () => {
	it('renders component', () => {
		const wrapper = shallow(<Icon icon={faCaretDown} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('handles click and calls provided callback', () => {
		const mockOnClick = jest.fn();
		const wrapper = mount(<Icon icon={faCaretDown} onClick={mockOnClick} />);

		wrapper.find('svg').simulate('click');
		expect(mockOnClick.mock.calls.length).toEqual(1);
	});
});
