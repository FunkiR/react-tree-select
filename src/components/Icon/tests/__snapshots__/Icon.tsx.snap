// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders component 1`] = `
<FontAwesomeIcon
  border={false}
  className="Icon"
  fixedWidth={false}
  flip={null}
  icon={
    Object {
      "icon": Array [
        320,
        512,
        Array [],
        "f0d7",
        "M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z",
      ],
      "iconName": "caret-down",
      "prefix": "fas",
    }
  }
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;

exports[`Icon component renders component correctly 1`] = `
<FontAwesomeIcon
  border={false}
  className="Icon"
  fixedWidth={false}
  flip={null}
  icon={
    Object {
      "icon": Array [],
      "iconName": "caret-down",
      "prefix": "fas",
    }
  }
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;
